cmake/CompilerConfiguration.cmake

```diff
@@ -0,0 +1,123 @@
+if(CMAKE_Fortran_COMPILER_ID STREQUAL "GNU")
+  set(CMAKE_Fortran_FLAGS
```

Comment:
I _think_ all of these `CMAKE_*_FLAGS` definitions will override any user-defined `CMAKE_*_FLAGS`. I would recommend either appending to the cache variables or again using targets and `target_compile_flags` (together with generator expressions for the various conditionals).

Reply:
I kept the initial version for the time being but I will work on this to make simpler.